src/ajax.js

```diff
@@ -462,14 +462,14 @@ jQuery.extend( {
 
 				// Raw string
 				getAllResponseHeaders: function() {
-					return state === 2 ? responseHeadersString : null;
+					return completed ? responseHeadersString : null;
 				},
 
 				// Caches the header
 				setRequestHeader: function( name, value ) {
-					var lname = name.toLowerCase();
-					if ( !state ) {
-						name = requestHeadersNames[ lname ] = requestHeadersNames[ lname ] || name;
+					if ( completed == null ) {
```

Comment:
I reserve those for cases where we explicitly want to differentiate `null` and `undefined` in order to keep the minified output more compressible. Each `=== undefined` costs about a byte.